examples/job_attachments/README.md

```diff
@@ -0,0 +1,198 @@
+
+Where does Job Attachments come into use in the Deadline Cloud job life cycle? 1) Job Submission, 2) Moving data to workers for job execution, 3) Copying job execution output back to the Job Attachments bucket and 4) Downloading Job Outputs.
+
+This README outlines CLIs and Python APIs to interface with each step in the Job Attachment process.
+
+Concepts:
+- Content Addressable Storage (CAS) - [wiki](https://en.wikipedia.org/wiki/Content-addressable_storage)
+
+## TLDR; Quick start
+
+### Creating a Job Attachments manifest "snapshot" to capture files for upload:
+```
+deadline manifest snapshot \
+    --root ./path/to/files \
+    --destination ./path/for/snapshot/manifest/ \
+    --name data-snapshot \
+    --include "*.txt" \                               (Optional)
+    --exclude "*.tmp"                                 (Optional)
+```
+
+### Uploading a manifest file to S3, useful for subsequent job submissions.
+```
+deadline manifest upload \
+    --farm-id farm-12345678123456781234567812345678 \
+    --queue-id queue-98765432987654329876543298765432 \
+    --s3-manifest-prefix where/to/store/manifests \                         (Optional)
+    --json \                                                                (Optional)
+    --profile \                                                             (Optional)
+    /path/for/snapshot/manifest/data-snapshot-2024-11-15T15-46-40.manifest
+```
+
+### Uploading files captured by Job Attachment "snapshot" to S3:
+```
+deadline attachment upload \
+    --manifests "/path/to/job/attachment.manifest" \
+    --s3-root-uri s3://my-queue/DeadlineCloud \
+    --json                                              (Optional)
+    --profile                                           (Optional)
+```
+
+### Visually show a "diff" of the current directory content against a previous manifest snapshot:
+```
+deadline manifest diff \
+    --root ./path/to/files \
+    --manifest /path/for/snapshot/manifest/data-snapshot-2024-11-15T15-46-40.manifest \
+    --include "*.txt" \                                   (Optional)
+    --exclude "*.tmp" \                                   (Optional)
+    --json                                                (Optional)
+```
+
+### Creating a "diff" manifest of the current directory content against a previous manifest snapshot:
+```
+deadline manifest snapshot \
+    --diff /path/for/snapshot/manifest/data-snapshot-2024-11-15T15-46-40.manifest \
+    --root ./path/to/files \
+    --destination ./path/for/snapshot/manifest/ \
+    --name data-snapshot-diff
+```
+
+### Downloading Job Attachment manifest for a Job (All job inputs), or Step (All job inputs and outputs of the step including dependencies)
+```
+deadline manifest download \
+    --farm-id farm-abcdabcdabcdabcdabcdabcdabcdabcd \
+    --queue-id queue-11111111111111111111111111111111 \
+    --job-id job-22222222222222222222222222222222 \
```

Comment:
These are optional too?

Reply:
farm, queue and job should not be optional? Manifest download keys on the farm, queue and job.